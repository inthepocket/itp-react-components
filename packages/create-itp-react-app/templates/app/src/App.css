body {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol';
}

.content {
  display: flex;
  justify-content: center;
  padding: 1em 0;
}