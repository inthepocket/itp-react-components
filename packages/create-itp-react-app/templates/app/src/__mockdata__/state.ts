import { FeatureInterface } from '<PROJECT-NAME>-core/types/Feature';
import { features } from './features';

interface EntitiesStateArgsInterface {
  collection: FeatureInterface[];
}

interface UiStateArgsInterface {
  collection: FeatureInterface[];
}

const entitiesState = (
  { collection = [] }: EntitiesStateArgsInterface,
) => collection.reduce(
  (state: any, object: any) => ({
    ...state,
    [object.id]: object,
  }),
  {},
);

const uiState = (
  { collection = [] } : UiStateArgsInterface,
) => ({
  items: collection.map((object: any) => object.id),
});

export default {
  entities: {
    features: entitiesState({ collection: features }),
  },
  ui: {
    features: {
      error: null,
      isFetchingFeatures: false,
      ...uiState({ collection: features }),
    },
  },
};
