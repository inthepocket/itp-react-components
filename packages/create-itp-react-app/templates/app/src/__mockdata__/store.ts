import { createStore } from 'redux';
import { default as rootReducer } from '<PROJECT-NAME>-core/reducers';

export const createMockStore = (mockedState: any) => {
  return createStore(rootReducer(), mockedState);
};
