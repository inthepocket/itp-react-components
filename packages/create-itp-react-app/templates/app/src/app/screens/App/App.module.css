@import '../../../hubble/variables.css';
@import '@inthepocket/itp-css/properties/breakpoints.css';

.App {
  position: relative;
  height: calc(100% - 2.5rem - 2.5rem);
  padding: 2.5rem;
  background-color: var(--colorBasePrimary100);
}

.App::before {
  position: absolute;
  right: 0;
  bottom: 0;
  z-index: 0;
  display: block;
  width: 55vw;
  height: 60vh;
  content: "";
  background-color: var(--colorBasePrimary900);
}

.content {
  margin: 2rem 0 0;
}

@media (--bp3) {
  .App {
    height: calc(100% - 3.75rem - 3.75rem);
    padding: 3.75rem;
  }
}
