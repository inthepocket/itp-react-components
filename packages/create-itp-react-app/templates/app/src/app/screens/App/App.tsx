import { default as React, useCallback, useEffect } from 'react';
import { useMappedState, useDispatch } from 'redux-react-hook';
import { fetchFeatures } from '<PROJECT-NAME>-core/actions/feature/featureActions';
import { getFeatureState } from '<PROJECT-NAME>-core/selectors/feature';
import Features from '<PROJECT-NAME>-common/components/molecules/Features/Features';
import Header from '<PROJECT-NAME>-common/components/molecules/Header/Header';
import { ReactComponent as Logo } from '<PROJECT-NAME>-common/assets/img/logo-itp-white.svg';
import { default as styles } from './App.module.css';

const useMappedFeaturesState = () => {
  const mappedState = useCallback(
    getFeatureState,
    ['features'],
  );

  return useMappedState(mappedState);
};

const App = () => {
  const dispatch = useDispatch();
  const featuresState: any = useMappedFeaturesState();
  const { features, isFetchingFeatures } = featuresState;

  useEffect(
    () => {
      dispatch(fetchFeatures.request());
    },
    featuresState,
  );

  return (
    <div className={styles.App}>
      <Logo className={styles.Logo} />
      <Header title="Welcome to your generated itp-react-app." />
      <Features features={features} isLoading={isFetchingFeatures} />
    </div>
  );
};

export default App;
