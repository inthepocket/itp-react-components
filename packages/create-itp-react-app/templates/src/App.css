body {
  font-family: sans-serif;
}

.app {

}
