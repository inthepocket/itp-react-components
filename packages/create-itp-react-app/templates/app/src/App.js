import React from 'react';
import 'normalize.css/normalize.css';
import './root.css';
import Header from '<PROJECT-NAME>-common/components/molecules/Header';
import Button from '<PROJECT-NAME>-common/components/atoms/Button';
import './App.css';

const App = () => (
  <div>
    <Header title="Create ITP React App" />
    <div className="content">
      <Button prefixCls="ant-btn" color="primary" type="primary" size="large">Large Button</Button>
    </div>
  </div>
);

export default App;

