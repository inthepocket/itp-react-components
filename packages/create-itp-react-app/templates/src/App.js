import React from 'react';
import 'normalize.css/normalize.css';
import './Root.css'
import Header from '<PROJECT-NAME>-common/components/molecules/Header';
import './App.css';

const App = () => (
  <div className="app">
    <Header title="Create ITP React App" />
  </div>
);

export default App;
