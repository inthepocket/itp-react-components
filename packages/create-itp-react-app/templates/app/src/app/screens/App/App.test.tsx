import * as React from 'react';
import reactTestRenderer from 'react-test-renderer';
import { StoreContext } from 'redux-react-hook';
import { default as mockState } from '<PROJECT-NAME>-mock/state';
import { createMockStore } from '<PROJECT-NAME>-mock/store';
import App from './App';

test('App snapshot: default', () => {
  const tree = reactTestRenderer
    .create((
      <StoreContext.Provider value={createMockStore(mockState)}>
        <App />
      </StoreContext.Provider>
    ))
    .toJSON();

  expect(tree).toMatchSnapshot();
});
