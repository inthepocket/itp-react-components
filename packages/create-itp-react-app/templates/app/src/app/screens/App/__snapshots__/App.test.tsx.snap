// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot: default 1`] = `
<div
  className="App"
>
  <svg
    className="Logo"
  >
    logo-itp-white.svg
  </svg>
  <div
    className="header"
  >
    <h1
      className="Title"
    >
      Welcome to your generated itp-react-app.
    </h1>
  </div>
  <p
    className="Features"
  >
    <span
      className="title"
    >
      Features:
    </span>
    <a
      className="Feature"
      href="https://github.com/facebook/create-react-app"
      target="_blank"
    >
      react-scripts
    </a>
    <a
      className="Feature"
      href="https://www.typescriptlang.org/"
      target="_blank"
    >
      Typescript
    </a>
    <a
      className="Feature"
      href="https://redux.js.org/"
      target="_blank"
    >
      redux
    </a>
    <a
      className="Feature"
      href="https://github.com/redux-saga/redux-saga"
      target="_blank"
    >
      redux-saga
    </a>
    <a
      className="Feature"
      href="https://github.com/facebookincubator/redux-react-hook"
      target="_blank"
    >
      redux-react-hook
    </a>
    <a
      className="Feature"
      href="https://github.com/piotrwitek/typesafe-actions"
      target="_blank"
    >
      typesafe-actions
    </a>
    <a
      className="Feature"
      href="https://github.com/css-modules/css-modules"
      target="_blank"
    >
      CSS Modules
    </a>
    <a
      className="Feature"
      href="https://github.com/inthepocket/hubble-mirror"
      target="_blank"
    >
      @inthepocket/hubble-mirror
    </a>
    <a
      className="Feature"
      href="https://github.com/inthepocket/itp-react-scripts/"
      target="_blank"
    >
      @inthepocket/itp-react-scripts
    </a>
    <a
      className="Feature"
      href="https://bitbucket.org/inthepocket/itp-css/"
      target="_blank"
    >
      @inthepocket/itp-css
    </a>
  </p>
</div>
`;

exports[`App snapshot: rerender to test useEffect hook 1`] = `
<div
  className="App"
>
  <svg
    className="Logo"
  >
    logo-itp-white.svg
  </svg>
  <div
    className="header"
  >
    <h1
      className="Title"
    >
      Welcome to your generated itp-react-app.
    </h1>
  </div>
  <p
    className="isLoading"
  >
    Loading features...
  </p>
</div>
`;

exports[`App snapshot: update tree to test useEffect hook 1`] = `
<div
  className="App"
>
  <svg
    className="Logo"
  >
    logo-itp-white.svg
  </svg>
  <div
    className="header"
  >
    <h1
      className="Title"
    >
      Welcome to your generated itp-react-app.
    </h1>
  </div>
  <p
    className="isLoading"
  >
    Loading features...
  </p>
</div>
`;
